src/snitch_append.cpp

```diff
@@ -1,65 +1,68 @@
 #include "snitch/snitch_append.hpp"
 
-#include <cinttypes> // for format strings
-#include <cstdio> // for std::snprintf
+#include <charconv> // for std::to_chars
 #include <cstring> // for std::memmove
 
```

Comment:
nit: I think we also need `<concepts>` for `std::floating_point`, `snitch_string.hpp` for `small_string_span`, `<type_traits>` for `std::is_same_v`, and `<system_error>` for `std::errc`